hoptimator-operator/src/main/java/com/linkedin/hoptimator/operator/subscription/SubscriptionReconciler.java

```diff
@@ -106,6 +150,102 @@ Pipeline pipeline(V1alpha1Subscription object) throws Exception {
+        owners.add(ownerReference);
+      }
+      obj.setMetadata(obj.getMetadata().ownerReferences(owners).resourceVersion(resourceVersion));
+      KubernetesApiResponse<DynamicKubernetesObject> response = operator.apiFor(obj).update(obj);
+      if (!response.isSuccess()) {
+        log.error("Error updating downstream resource {}/{}: {}.", namespace, name, response.getStatus().getMessage());
+        return false;
+      }
+    } else {
+      log.info("Creating downstream resource {}/{} as \n{}", namespace, name, yaml);
+      obj.setMetadata(obj.getMetadata().addOwnerReferencesItem(ownerReference));
+      KubernetesApiResponse<DynamicKubernetesObject> response = operator.apiFor(obj).create(obj);
+      if (!response.isSuccess()) {
+        log.error("Error creating downstream resource {}/{}: {}.", namespace, name, response.getStatus().getMessage());
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean checkStatus(String yaml) {
+    DynamicKubernetesObject obj = Dynamics.newFromYaml(yaml);
+    String namespace = obj.getMetadata().getNamespace();
+    String name = obj.getMetadata().getName();
+    String kind = obj.getKind();
+    try {
+      KubernetesApiResponse<DynamicKubernetesObject> existing = operator.apiFor(obj).get(namespace, name);
+      existing.onFailure((code, status) -> log.warn("Failed to fetch {}/{}: {}.", kind, name, status.getMessage()));
+      if (!existing.isSuccess()) {
+        return false;
+      }
+      if (isReady(existing.getObject())) {
+        log.info("{}/{} is ready.", kind, name);
+        return true;
+      } else {
+        log.info("{}/{} is NOT ready.", kind, name);
+        return false;
+      }
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  private static boolean isReady(DynamicKubernetesObject obj) {
+    // We make a best effort to guess the status of the dynamic object. By default, it's ready.
+    if (obj == null || obj.getRaw() == null) {
+      return false;
+    }
+    try {
+      return obj.getRaw().get("status").getAsJsonObject().get("ready").getAsBoolean();
+    } catch (Exception e) {
+      log.debug("Exception looking for .status.ready. Swallowing.", e);
+    }
+    try {
+      return obj.getRaw().get("status").getAsJsonObject().get("state").getAsString()
+        .matches("(?i)READY|RUNNING|FINISHED");
+    } catch (Exception e) {
+      log.debug("Exception looking for .status.state. Swallowing.", e);
+    }
+    try {
+      return obj.getRaw().get("status").getAsJsonObject().get("jobStatus").getAsJsonObject()
+        .get("state").getAsString().matches("(?i)READY|RUNNING|FINISHED");
+    } catch (Exception e) {
+      log.debug("Exception looking for .status.jobStatus.state. Swallowing.", e);
+    }
+    // TODO: Look for common Conditions
+    log.warn("Resource {}/{}/{} considered ready by default.", obj.getMetadata().getNamespace(),
```

Comment:
should this be warn or info?

Reply:
[Minor] general comment, I see log have just plain value, would it be possible to have key={value} format? so if one knows the key they can search for that? Here, it seems like I need to know the value, may be we do know that, however, it would be more readable if it's in key={value} format.

Reply:
It's warn cuz it could indicate a problem. We're reporting success just on the basis that 1) the resource exists, and 2) the status didn't look like one we can interpret. Ideally we'd update this code anytime we see a warning here.